packages/nimble-blazor/NimbleBlazor.Acceptance.Tests/Tests/AcceptanceTestsBase.cs

```diff
@@ -0,0 +1,62 @@
+﻿using Microsoft.Playwright;
+using Xunit;
+
+namespace NimbleBlazor.Tests.Acceptance
+{
+    [Collection(nameof(PlaywrightFixture))]
+    public abstract class AcceptanceTestsBase : IClassFixture<BlazorServerWebHostFixture>
+    {
+        private PlaywrightFixture _playwrightFixture;
+        private readonly BlazorServerWebHostFixture _blazorServerClassFixture;
+
+        protected AcceptanceTestsBase(
+            PlaywrightFixture playwrightFixture,
+            BlazorServerWebHostFixture blazorServerClassFixture)
+        {
+            _playwrightFixture = playwrightFixture;
+            _blazorServerClassFixture = blazorServerClassFixture;
+        }
+
+        private IBrowserContext BrowserContext
+        {
+            get
+            {
+                return _playwrightFixture.BrowserContext!;
+            }
+        }
+
+        protected async Task<AsyncDisposablePage> NewPageForRouteAsync(string route)
+        {
+            var page = await BrowserContext.NewPageAsync();
+            await NavigateToPageAsync(page, route);
+            await WaitForNimbleBlazorInitializationAsync(page);
+            return new AsyncDisposablePage(page);
+        }
+
+        private async Task NavigateToPageAsync(IPage page, string route)
+        {
+            var address = new Uri(_blazorServerClassFixture.ServerAddress!, route).AbsoluteUri;
+            await page.GotoAsync(address);
+        }
+
+        private async Task WaitForNimbleBlazorInitializationAsync(IPage page)
+        {
+            await page.WaitForFunctionAsync("window.NimbleBlazor && window.NimbleBlazor.calledAfterStarted === true");
```

Comment:
Note: I ran into some test intermittency before I added this logic to wait for Nimble Blazor initialization. For example, for the dialog test, Playwright could click the button immediately after the page loaded (but `window.NimbleBlazor.calledAfterStarted` was `true`), and in that scenario, the button click ended up being a no-op (didn't open the dialog as expected).  
I think it might be theoretically possible for an end-user to encounter the same issue, but unlikely.